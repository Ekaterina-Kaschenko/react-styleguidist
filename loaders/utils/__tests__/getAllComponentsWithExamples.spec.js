import getAllComponentsWithExamples from '../getAllComponentsWithExamples';

const sections = [
	{
		name: 'Readme',
		content: 'Readme.md',
		components: [],
		sections: [],
	},
	{
		name: 'Components',
		components: [
			{
				filepath: 'components/Button/Button.js',
				examples: 'require()',
			},
			{
				filepath: 'components/Icon/Icon.js',
			},
		],
		sections: [],
	},
	{
		name: 'Nesting',
		components: [],
		sections: [
			{
				name: 'Nested',
				components: [
					{
						filepath: 'components/Image/Image.js',
					},
					{
						filepath: 'components/Modal/Modal.js',
						examples: 'require()',
					},
				],
				sections: [],
			},
			{
				name: 'Nested 2',
				components: [
					{
						filepath: 'components/Avatar/Avatar.js',
					},
				],
				sections: [],
			},
		],
	},
];

it('should return only components with example file', () => {
	const result = getAllComponentsWithExamples(sections);
	expect(result).toMatchSnapshot();
});
