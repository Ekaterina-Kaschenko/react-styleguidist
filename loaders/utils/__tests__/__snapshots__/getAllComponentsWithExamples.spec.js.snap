exports[`test should return only components with example file 1`] = `
Array [
  Object {
    "examples": "require()",
    "filepath": "components/Button/Button.js",
  },
  Object {
    "examples": "require()",
    "filepath": "components/Modal/Modal.js",
  },
]
`;
